tokens/tsm-report.go

```diff
@@ -0,0 +1,102 @@
+package tokens
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/fxamacker/cbor/v2"
+)
+
+const (
+	TSMReportMediaTypeCBOR = "application/vnd.veraison.tsm-report+cbor"
+	TSMReportMediaTypeJSON = "application/vnd.veraison.tsm-report+json"
+)
+
+// BinaryString is base64url (§5 of RFC4648) without padding.
+type BinaryString []byte
+
+// MarshalJSON encodes BinaryString as JSON
+func (o BinaryString) MarshalJSON() ([]byte, error) {
+	return json.Marshal(
+		base64.RawURLEncoding.EncodeToString(o),
+	)
+}
+
+// UnmarshalJSON decodes BinaryString from JSON
+func (o *BinaryString) UnmarshalJSON(b []byte) error {
+	var s string
+
+	if err := json.Unmarshal(b, &s); err != nil {
+		return err
+	}
+
+	decoded, err := base64.RawURLEncoding.DecodeString(s)
+	if err != nil {
+		return err
+	}
+
+	*o = decoded
+	return nil
+}
+
+/* TSMReport represents the TSM report generated by Linux
+ *   see docs/tsm-report.cddl for definition
+ */
```

Comment:
Unless there’s a good reason to diverge from the convention, I’d use `//`

```suggestion
// TSMReport represents the TSM report generated by Linux
// See docs/tsm-report.cddl for definition
```